### PR TITLE
odim-98: Get response on session service gives 401 status code error with invalid basic authentication

### PR DESCRIPTION
**Pull request by rameshkk-hpe**
Issue:

1.Execute Get operation on  the session service with invalid basic authentication   ,It gives 401 unauthorized error

GET API: https://<odimra ip>:45000/redfish/v1/SessionService

Basic authentication: admin01/HP!nvent01

2. Expected Result: It should execute with 200 status code

Developer Note:
1) Fixed for URL /redfish/v1 as well as it is a common issue impacting URLs which do not need basic auth
2) Please enable "Hide Whitespace change" while checking the difference as a huge chunk of code had to be included in a new if condition.

### CODE REVIEW

File: svc-api/main.go [hunk 1]
```diff
@@ -48,6 +48,21 @@ func main() {
 		}
 		basicAuth := r.Header.Get("Authorization")
 		var basicAuthToken string
+		// This if block added to fix issue odim-98
+		if basicAuth != "" {
```
Comment (a-ajith) [inline svc-api/main.go:51]: same if condition check is happening in line number 66 also. please consolidate them.

**Review by a-ajith — COMMENTED** [review]
Address review comments

File: svc-api/main.go [hunk 2]
```diff
@@ -48,6 +48,21 @@ func main() {
...
+			var URLs_not_requiring_basic_auth = []string {"/redfish/v1","/redfish/v1/SessionService"}
```
Comment (a-ajith) [inline svc-api/main.go]: private variable names should start with a small letter. In Golang we use camel casing as a standard. Try to use meaningful short names.

File: svc-api/main.go [hunk 3]
```diff
@@ -48,6 +48,21 @@ func main() {
...
+			var found bool = false
+			for _, item := range URLs_not_requiring_basic_auth {
+				if item == path {
+					found = true
+					break
+				}
+			}
+			if found {
```
Comment (a-ajith) [inline svc-api/main.go]: This if condition is not required. body of this can be replaced at line 57

File: svc-api/main.go (hunk 1 above)
Comment (rameshkk-hpe) [reply svc-api/main.go:51]: Done.
Please enable "Hide whitespace changes" while checking the code difference as a huge chunk of existing code had to be included in a new if condition (and hence indented).

File: svc-api/main.go (hunk 2 above)
Comment (rameshkk-hpe) [reply svc-api/main.go]: Done.

File: svc-api/main.go (hunk 3 above)
Comment (rameshkk-hpe) [reply svc-api/main.go]: Done.

**Review by a-ajith — APPROVED** [review]

**Review by shivacharanms — APPROVED** [review]

**Pull request merged**